app/controllers/application_controller.rb

```diff
@@ -13,7 +13,18 @@ def after_sign_in_path_for(resource)
   private
 
   # we can enable this for everybody, it's hidden by default
-  def enable_rack_mini_profiler
-    Rack::MiniProfiler.authorize_request
+  def rack_mini_profiler
+    show =
+      if params.key?(:enable_rack_mini_profiler)
+        cookies[:rack_mini_profiler] = "show"
+        true
+      elsif params.key?(:disable_rack_mini_profiler)
+        cookies.delete(:rack_mini_profiler)
+        false
+      end
+
+    if show || cookies[:rack_mini_profiler]
+      Rack::MiniProfiler.authorize_request
+    end
```

Comment:
I tried doing a one-liner here but sometimes the widget didn't show up, no idea what was going on there, so I used a normal if / end